site/docs/components/autocomplete.mdx

```diff
@@ -42,11 +42,17 @@ Autocomplete is forgiving, letting people make mistakes such as spelling errors
     - For a limited dataset you might update autocomplete options with new matching results on every key stroke the user types.
     - For large datasets you might delay updating options until there's at least some number of characters (usually 3). If so, ensure results of only 1 or 2 characters long can be shown.
 
+
 ### Matching results
 
 - **Fuzzy search**: a user can write some text and a partial match returns results. Consider whether the text needs to be in order, what effect spaces have, and the effect of capitalization on the match. For example, a forgiving search input that lets you type “design” to match the result “Marketing Design” may be ideal for users that aren't sure what exactly it will be called here.
 - **Smart capitalization matching**: consider appropriate capitalization matching for the data available. For example, when people type a search query using lowercase letters, you might return all, case-insensitive results and when they type a search query containing capital letters, you might return only case-sensitive results matching their capitalization.
 
+### Displaying results
+
+- **Searchable selects**: a user can write some text and potential autocomplete matches or results are shown in a dropdown. The user may either keep typing, or select a result from the dropdown, wherein that selection sets the value of the element. This pattern should be used when there is a very large set of predefined values, such as choosing employee names from a list of all employees.
+- **Immediate results matching**: a user can write some text and potential autocomplete matches or results are filtered live and shown on the page itself as the user types. These results will be further matched and refined as the user continues to type. The user may either keep typing, or select the desired result that has appeared on the page. This pattern should be used when there is a smaller set of initial predefined values, and when seeing the results may help the user to distinguish between them, such as searching for blog posts or articles.
```

Comment:
I don't understand what is meant by "should be used when there is a smaller set of initial predefined values" - would you not use this pattern if there were many values to match from?

Reply:
@zakwash01 Yep, that's what I was thinking - I was thinking that if the list were very large, you would be better off using a searchable select and then a complete reload/results page to view what you've filtered down to. 

I mainly have concerns due to load issues, e.g. if you're trying to live filter a list of 100+ large objects, will the filtering of the list keep up with your typing speed? I also feel that live filtering a very large list of objects can be overwhelming for users - _but_ we do have a few examples in platform where we are using this pattern for large lists, e.g. searching for users or comments.